atd-vze/src/Components/GridTable.js

```diff
@@ -516,22 +526,19 @@ const GridTable = ({
                   )}
                 </ButtonGroup>
               </ButtonToolbar>
-              <Table responsive>
-                <GridTableHeader
-                  query={query}
-                  handleTableHeaderClick={handleTableHeaderClick}
-                  sortColumn={sortColumn}
-                  sortOrder={sortOrder}
-                />
-
-                <tbody>
-                  {loading ? (
-                    <Spinner className="mt-2" color="primary" />
-                  ) : (
-                    data && dataEntries
-                  )}
-                </tbody>
-              </Table>
+              {loading ? (
+                <Spinner className="mt-2" color="primary" />
+              ) : (
+                <Table responsive>
+                  <GridTableHeader
+                    query={query}
+                    handleTableHeaderClick={handleTableHeaderClick}
+                    sortColumn={sortColumn}
+                    sortOrder={sortOrder}
+                  />
+                  <tbody>{data && dataEntries}</tbody>
+                </Table>
+              )}
```

Comment:
Also cool stuff, I think it makes sense to have the spinner encompass the page and not show headers until it loads everything.